mesheryctl/internal/cli/root/perf/apply.go

```diff
@@ -261,14 +278,45 @@ var applyCmd = &cobra.Command{
 	},
 }
 
+func multipleProfileConfirmation(profiles []models.PerformanceProfile) int {
+	reader := bufio.NewReader(os.Stdin)
+
+	for index, a := range profiles {
+		fmt.Printf("Index: %v\n", index)
+		fmt.Printf("Name: %v\n", a.Name)
+		fmt.Printf("ID: %s\n", a.ID.String())
+		fmt.Printf("Endpoint: %v\n", a.Endpoints[0])
+		fmt.Printf("Load Generators: %v\n", a.LoadGenerators[0])
+		fmt.Println("---------------------")
+	}
+
+	for {
+		fmt.Printf("Enter the index of profile?? ")
+		response, err := reader.ReadString('\n')
+		if err != nil {
+			log.Fatal(err)
+		}
+		response = strings.ToLower(strings.TrimSpace(response))
+		index, err := strconv.Atoi(response)
+		if err != nil {
+			log.Info(err)
+		}
+		if index < 0 || index >= len(profiles) {
+			log.Info("Invalid index")
+		} else {
+			return index
+		}
+	}
+}
+
 func init() {
-	applyCmd.Flags().StringVar(&testURL, "url", "", "(required) Endpoint URL to test")
+	applyCmd.Flags().StringVar(&testURL, "url", "", "(required/optional) Endpoint URL to test")
```

Comment:
is this correct? i'm not sure how a flag can be required and optional?

Reply:
The flag is required if you want to run the test with the given URL, or you want to use the one that's stored in the profile itself.

Reply:
hmm,  @leecalcote  might be in a better position to answer this?